screens/ChooseLocationScreen.tsx

```diff
@@ -0,0 +1,274 @@
+import * as LocationProvider from 'expo-location';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { Image, StyleSheet, TouchableOpacity, View } from 'react-native';
+import MapView, { EdgeInsets, Marker, Region } from 'react-native-maps';
+import {
+  ActivityIndicator,
+  Caption,
+  FAB,
+  TextInput,
+  useTheme,
+} from 'react-native-paper';
+import { Theme } from 'react-native-paper/lib/typescript/types';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import { useNavigation } from '@react-navigation/native';
+import currentLocationIcon from '../assets/current-location.png';
+import { SecondaryAppbar } from '../components/SecondaryAppbar';
+import { MapContext } from '../context';
+import { findCurrentLocation, findPlace } from '../context/MapContext';
+import { getHotspotMarker } from '../models/Hotspot';
+import {
+  defaultLocation,
+  getFormattedAddress,
+  Location,
+} from '../models/Location';
+import { RootStackScreenProps } from '../types';
+import { loadHotspots } from '../utils/hotspots';
+
+const intialRegion: Region = {
+  latitude: 46.77293258116839,
+  longitude: 23.587688864363546,
+  latitudeDelta: 0.0922,
+  longitudeDelta: 0.0421,
+};
+
+export const ChooseLocationScreen = ({
+  route,
+}: RootStackScreenProps<'ChooseLocation'>) => {
+  const { hotspots, setHotspots } = useContext(MapContext);
+  const [selectedLocation, setSelectedLocation] = useState<Location>();
+
+  const canConfirmAddress =
+    selectedLocation &&
+    selectedLocation!.latitude != 0 &&
+    selectedLocation!.longitude != 0;
+
+  const mapRef = useRef<MapView>(null);
+  const navigation = useNavigation();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const searchForAddress = async (address: string): Promise<void> => {
+    try {
+      const [{ latitude, longitude }] = await LocationProvider.geocodeAsync(
+        address
+      );
+
+      animateMapToRegion({ latitude: latitude, longitude: longitude });
+    } catch (e) {
+      alert('Address not found');
+    }
+  };
+
+  const insets = useSafeAreaInsets();
+  const theme = useTheme();
+  const styles = getStyles(theme, insets);
+
+  const animateMapToRegion = (_location: Location) => {
+    mapRef.current?.animateToRegion({
+      latitude: _location.latitude,
+      longitude: _location.longitude,
+      latitudeDelta: 0,
+      longitudeDelta: 0,
+    });
+  };
+
+  useEffect(() => {
+    setSelectedLocation(route.params.location);
+    if (route.params.location) {
+      animateMapToRegion(route.params.location);
+    }
+  }, [route.params.location]);
+
+  useEffect(() => {
+    const load = async () => {
+      setIsLoading(true);
+      const { success, hotspots = [] } = await loadHotspots();
+      setIsLoading(false);
+      if (!success) alert('Failed to load hotspots');
+      setHotspots(hotspots);
+    };
+    load();
+  }, []);
```

Comment:
this effect is not needed here, the hotspots should already be loaded from the server by the time the user gets here, so using the MapContext should be enough

Reply:
Right. I removed it.